File: grid.py
from pyprocessing import *
import defs

grid = []

def initGrid():
    for i in range(0,defs.GRID_COUNT_X):
        grid.append([])
        for j in range(0,defs.GRID_COUNT_Y):
            space = Space(i,j,None)
            grid[i].append(space)
    return grid

def drawGrid():
    for i in range(0,defs.GRID_COUNT_X):
        for j in range(0,defs.GRID_COUNT_Y):
            # print grid[i][j]
            grid[i][j].display()

class Space:
    def __init__(self,x,y,piece):
        self.x = x
        self.y = y
        self.piece = piece
        self.filled = False
        self.px = defs.GRID_OFFSET_X + (self.x * defs.GRID_SPACE_SIZE)
        self.py = defs.GRID_OFFSET_Y + (self.y * defs.GRID_SPACE_SIZE)
    def display(self):
        if self.piece is None:
            fill(255,255,255)
        else:
            self.piece.setColor()
        rect(self.px,self.py,defs.GRID_SPACE_SIZE,defs.GRID_SPACE_SIZE)
    def setPiece(self,piece):
        self.piece = piece
        if piece is None:
            self.filled = False
        else:
            self.filled = True
        # print "X:%d Y:%d Filled:%s" % (self.x,self.y,self.filled)

File: intellij/watertetris/src/Grid.java
/**
 * Created by Scott on 5/5/2014.
 */
class Grid {
    static Space[][] GRID;

    public static void initGrid()
    {
        GRID = new Space[defs.GRID_COUNT_X][defs.GRID_COUNT_Y];

        for(int i =0;i<defs.GRID_COUNT_X;i++)
        {
            for(int j = 0;j<defs.GRID_COUNT_Y;j++)
            {
                GRID[i][j] = new Space(i,j,null);
            }
        }
    }

    public static void drawGrid()
    {
        for(int i =0;i<defs.GRID_COUNT_X;i++)
        {
            for(int j = 0;j<defs.GRID_COUNT_Y;j++)
            {
                GRID[i][j].display();
            }
        }
    }

    public static boolean offGrid(int x, int y)
    {
        return !(x < defs.GRID_COUNT_X && y < defs.GRID_COUNT_Y && x >= 0);
    }
}
